linera-execution/src/wasm/runtime_api.rs

```diff
@@ -17,22 +17,19 @@ use linera_witty::{wit_export, Instance, RuntimeError};
 use tracing::log;
 
 use super::WasmExecutionError;
-use crate::{
-    BaseRuntime, BytecodeId, ContractRuntime, ContractSyncRuntimeHandle, ExecutionError,
-    ServiceRuntime, ServiceSyncRuntimeHandle,
-};
+use crate::{BaseRuntime, BytecodeId, ContractRuntime, ExecutionError, ServiceRuntime};
 
 /// Common host data used as the `UserData` of the system API implementations.
-pub struct SystemApiData<Runtime> {
+pub struct RuntimeApiData<Runtime> {
     runtime: Runtime,
     active_promises: HashMap<u32, Box<dyn Any + Send + Sync>>,
     promise_counter: u32,
 }
 
-impl<Runtime> SystemApiData<Runtime> {
-    /// Creates a new [`SystemApiData`] using the provided `runtime` to execute the system APIs.
+impl<Runtime> RuntimeApiData<Runtime> {
+    /// Creates a new [`RuntimeApiData`] using the provided `runtime` to execute the system APIs.
     pub fn new(runtime: Runtime) -> Self {
-        SystemApiData {
+        Self {
```

Comment:
Nit: I usually use `Self` in the method signatures to keep them short, and the expanded name when constructing to avoid using `Self` too much and making it harder to find what it `Self` when glancing at the code.

```suggestion
        RuntimeApiData {
```

Reply:
I usually avoid using the non syntactic sugared full name because when refactoring, changing the type's name would require many consequent changes. But I guess this could be done for the constructor.